universal-application-tool-0.0.1/app/services/applicant/question/NumberQuestion.java

```diff
@@ -27,20 +27,27 @@ public boolean hasQuestionErrors() {
       return ImmutableSet.of();
     }
 
-    NumberQuestionDefinition definition = getQuestionDefinition();
-    long answer = getNumberValue().get();
+    NumberQuestionDefinition questionDefinition = getQuestionDefinition();
+
+    // If there is no minimum value set, accept a blank answer.
+    if (getNumberValue().isEmpty() && questionDefinition.getMin().isEmpty()) {
+      return ImmutableSet.of();
+    }
+
     ImmutableSet.Builder<ValidationErrorMessage> errors = ImmutableSet.builder();
 
-    if (definition.getMin().isPresent()) {
-      long min = definition.getMin().getAsLong();
-      if (answer < min) {
+    if (questionDefinition.getMin().isPresent()) {
+      long min = questionDefinition.getMin().getAsLong();
+      // If the value is empty, consider it to be "less than the minimum".
+      if (getNumberValue().isEmpty() || getNumberValue().get() < min) {
         errors.add(ValidationErrorMessage.numberTooSmallError(min));
       }
     }
 
-    if (definition.getMax().isPresent()) {
-      long max = definition.getMax().getAsLong();
-      if (answer > max) {
+    if (questionDefinition.getMax().isPresent()) {
+      long max = questionDefinition.getMax().getAsLong();
+      // If the value is empty, do not consider it to be "greater than the maximum".
+      if (getNumberValue().isPresent() && getNumberValue().get() > max) {
```

Comment:
I think I would invert this - consider empty similar to the way that NaN works, which is to say, it fails any test you give it - it's both below the minimum and above the maximum.

Reply:
Here's my thinking on it: If someone leaves a number answer blank, it implies "none". None is below the maximum (unless then max is negative...).

I can see either way though.

Reply:
OK, updated!